airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/default_file_based_stream.py

```diff
@@ -65,17 +72,49 @@ def read_records_from_slice(self, stream_slice: StreamSlice) -> Iterable[Mapping
         for file in stream_slice["files"]:
             # only serialize the datetime once
             file_datetime_string = file.last_modified.strftime("%Y-%m-%dT%H:%M:%SZ")
+            n_skipped = line_no = 0
+
             try:
                 for record in parser.parse_records(file, self._stream_reader):
+                    line_no += 1
+
                     if not self.record_passes_validation_policy(record):
-                        logging.warning(f"Record did not pass validation policy: {record}")
+                        n_skipped += 1
                         continue
                     record[self.ab_last_mod_col] = file_datetime_string
                     record[self.ab_file_name_col] = file.uri
                     yield stream_data_to_airbyte_message(self.name, record)
                 self._cursor.add_file(file)
+
+            except StopSyncPerValidationPolicy:
+                yield AirbyteMessage(
```

Comment:
Do we need a new status for connections that are "paused"?

Setting the sync as successful feels misleading since the data wasn't replicated.

non-blocking since we have a similar issue with other sources. [We do something similar for salesforce when we hit a rate limit](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-salesforce/source_salesforce/source.py#L151-L153)

Reply:
> do you know whether related code changes would be isolated to the CDK?

I think we would need to update [the protocol](https://github.com/airbytehq/airbyte-protocol/blob/main/protocol-models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml#L250-L253) and the platform to handle the new status

> would it be sufficient to change the level to Level.Error?
The platform doesn't use the log level to determine the status. We would want to raise an `AirbyteTracedException`

Reply:
Ah, right. The `INCOMPLETE` status feels right to me. WDYT?